solana-library/spl_token.sol

```diff
@@ -43,7 +43,7 @@ library SplToken {
 	/// @param account the token account where the minted tokens should go
 	/// @param authority the public key of the mint authority
 	/// @param amount the amount of tokens to mint
-	function mint_to(address mint, address account, address authority, uint64 amount) internal {
+	function mint_to(address mint, address account, address authority, uint64 amount) view internal {
```

Comment:
This is wrong. minting can't be a view operation. None of these should be view

Reply:
Solang raises a warning if the `view` keyword is missing:

```
warning: function can be declared 'view'
   ┌─ /Users/lucas/Documents/solang/solana-library/spl_token.sol:46:2
   │
46 │     function mint_to(address mint, address account, address authority, uint64 amount) internal {
   │
```

Reply:
`view` means that state cannot be changed and this is a read-only function which can run without a transaction.

Clearly minting changes state.

Reply:
The language server also warns that all these functions should be view.

Reply:
Now, the warning is a different matter. The problem the compiler has is that it cannot know whether a function called view `.call()` changes state. The same thing exists with solc.

I'm not sure how to solve this. For Solana it might be possible to talk the list of meta accounts and seeing if any of them are writable. However that doesn't solve the problem for Substrate, and also it might not always be possible to walk the list of meta accounts on Solana eiher, if they're passed in via a function.

Some new syntax might be needed for this.

Alternatively, we could do something like "if `.call()` is done, then don't warn about `view` being possible". I Think that might make the most sense.

Reply:
The warning makes the developer believe he/she can make the function `view`. Hardly ever will someone doubt the compiler is wrong. If the warning is not 100% accurate, then we should not have raised it.

Reply:
Feel free to raise an issue